src/cli/subcommands/car_cmd.rs

```diff
@@ -122,12 +115,16 @@ mod tests {
             car
         }
 
-        fn into_stream(self) -> impl Stream<Item = BlockPair> {
-            futures::stream::iter(self.0.into_iter().map(|b| (b.cid, b.data)))
+        fn into_stream(self) -> impl Stream<Item = std::io::Result<super::Block>> {
+            futures::stream::iter(
+                self.0
+                    .into_iter()
+                    .map(|Block { cid, data }| Ok::<_, std::io::Error>(super::Block { cid, data })),
```

Comment:
We map block into block?

Reply:
From `fvm_ipld_car::Block` to `crate::utils::db::car_stream::Block`

Reply:
Why use `fvm_ipld_car::Block`?

Reply:
Need to use `CarHeader` API to convert it to car bytes. I don't find equivalent API in our own `CarHeader` tho

```rust
async fn into_car_bytes(self) -> Vec<u8> {
            // Dummy root
            let writer = CarHeader::from(vec![self.0[0].cid]);
            let mut car = vec![];
            let mut stream = pin!(futures::stream::iter(self.0).map(|b| (b.cid, b.data)));
            writer
                .write_stream_async(&mut car, &mut stream)
                .await
                .unwrap();
            car
        }
```

Reply:
Then let's implement our own CAR sink. Building on top of `CarHeader` is no use.

Reply:
Or simply generate ForestCAR.zst bytes.

Reply:
Nvm, actually it's OK to use own `Block` by updating the roundtrip test, fixed.